airflow/migrations/versions/0a2a5b66e19d_add_task_reschedule_table.py

```diff
@@ -66,13 +66,14 @@ def upgrade():
         sa.Column('id', sa.Integer(), nullable=False),
         sa.Column('task_id', sa.String(length=250), nullable=False),
         sa.Column('dag_id', sa.String(length=250), nullable=False),
-        # use explicit server_default=None otherwise mysql implies defaults for first timestamp column
-        sa.Column('execution_date', timestamp(), nullable=False, server_default=None),
+        sa.Column('execution_date', timestamp(), nullable=False,
+                  server_default=sa.text('CURRENT_TIMESTAMP(6)')),
         sa.Column('try_number', sa.Integer(), nullable=False),
-        sa.Column('start_date', timestamp(), nullable=False),
-        sa.Column('end_date', timestamp(), nullable=False),
+        sa.Column('start_date', timestamp(), nullable=False, server_default=sa.text('CURRENT_TIMESTAMP(6)')),
+        sa.Column('end_date', timestamp(), nullable=False, server_default=sa.text('CURRENT_TIMESTAMP(6)')),
```

Comment:
Additionally: I am not sure this is the right fix: we don't want the default value of these columns to be "Now()" or any variation of it.

Reply:
I'm not a Mysql expert by any means, but I understand the issue to be that if you specify a timestamp to column to be not null you have to supply a default value.  Given the model also knows that it's not null it *should* be the case that we never have to rely on the database's default?

Is the current_timestamp better or worse that any other default that could be used at the database level?

Maybe another option is to require mysql to use this flag? https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_explicit_defaults_for_timestamp

No idea how feasible that would be.